doc/src/EntitySpecMeta.dox

```diff
@@ -15,177 +15,309 @@
+ * should be presented. The correct trait collection to use would be
+ * just `blob` and `scene`.
+ *
+ * If it was required that the data be in a specific format (eg: USD)
+ * then the `mimeType` property defined by the `blob` trait should be
+ * set accordingly.
+ *
+ * If a host _did_ however require that a scene file contained a certain
+ * type of data  - a camera for example a camera loader, then the
+ * `camera` trait should be added to indicate that only scenes
+ * containing one should be presented.
+ *
+ * Another example would be the when quering for an an EDL file in
+ * relation to a sequence entity. The traits used should only include a
+ * description of the nature of any of its content if this is
+ * specifically required for its use, so typically `blob`, `EDL` is all
+ * that would be needed.
+ *
+ * @subsection Publishing
+ *
+ * When populating specifications or trait collections for publishing a
+ * similar principal applies. The only relevant traits are the ones that
+ * cover the data that is registed, the API ensure this as registration
+ * requires the entity data to be passed via specification.
+ *
+ * @subsection Descriptions
+ *
+ * When a manager populates the @needsref entityTraits response, then
+ * it should return traits that describe the entity. This is perhaps
+ * the only main exception to guidance above. The result of this query
+ * will be used by a host to decide if it is capable of consuming the
+ * entity's data in a specific way, so including all applicable traits
+ * is generally benefical.
+ *
+ * @note This may need a little time to settle in paractice. There has
+ * been discussion around whether there is need to define 'excluded'
+ * traits, or how best to handle situations such as the 'USD camera
+ * cache' example, where there may also be caches with meshes _and_
+ * cameras. How does that intersection get handled? One option may be
+ * that the `camera` trait also included a `path` property that allowed
+ * the rest of the scene to be filtered out...
+ *
+ * @section specification_mapping Mapping to Native Types
  *
  * Where possible, both hosts and managers should attempt to map
- * well-known specifications to their native types. A manager with no UI
- * of its own may not need to, but in most situations, a consistent
- * mapping of specifications to first-class concepts is essential for a
+ * well-known trait collections to their native types. A manager with no
+ * UI of its own may not need to, but in most situations, a consistent
+ * mapping of traits to first-class concepts is essential for a
  * well-formed user experience.
  *
- * For example, a compositor may map an ImageSequenceSpecification to
- * its native image read and write nodes, and a manager map the same to
- * its own supported clip format so transcodes are generated and
- * displayed in its browser.
+ * For example, a compositor may map the traits of the
+ * `ImageSequenceSpecification` to its native image read and write
```

Comment:
[question] I'm struggling to understand this a bit. How does a trait map to a read/write node?  How is such a mapping used?  Looking at the text it replaced, it's just as mysterious to me, so not an issue introduced by this PR ;)

Reply:
[This](https://github.com/TheFoundryVisionmongers/OpenAssetIO/pull/249/commits/94d3a39292e2d378beef3e9449fef188c44c6eb0) any better?

Reply:
Updated [here](https://github.com/TheFoundryVisionmongers/OpenAssetIO/pull/249/commits/b545a8153186737909df9ec76382067256bbdfce) too.

Reply:
Definitely improved, but then broken again, added a separate comment re. that, so I'll resolve this one.